### PR TITLE
Feature/fix xsd path loading and creation of XML from relative xpath of field

### PR DESCRIPTION
**Pull request by rouschr**
This fixes a problem that has crept in, probably during a refactoring to use a new core (sdk downloader) version.
The way the sdkRootFolder was used was wrong for the NoticeXmlTagSorter, the sorter needs a specific SDK folder and not the root folder where all SDKs are located. We had a similar issue before in another project.
In the unit test this step is not covered as there we directly pass a dummy SDK folder.

In addition to this, the relative xpath of fields, should be fully rebuilt and never try to reuse existing elements.
This fixes an issue in the XML generation for some fields that have intermediary nodes in the relative xpath.
All unit tests still pass.

### CODE REVIEW

File: src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java [hunk 1]
```diff
@@ -489,10 +494,15 @@ private static void buildFields(final Document doc, final FieldsAndNodes fieldsA
       // XML.
       final boolean isAttribute = tagOrAttr.startsWith("@") && tagOrAttr.length() > 1;
 
-      final Optional<NodeList> foundElementsOpt = !isAttribute && xpathExprOpt.isPresent()
-          ? Optional.of(XmlUtils.evaluateXpathAsNodeList(xpathInst, previousElem,
-              xpathExprOpt.get(), fieldId))
-          : Optional.empty();
+      final Optional<NodeList> foundElementsOpt;
+      if (!isAttribute && xpathExprOpt.isPresent()) {
```
Comment (rousso) [inline src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java]: ?! Both branches do the same thing here. What is the if for?

File: src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java [hunk 2]
```diff
@@ -661,7 +671,7 @@ public static XPath setXmlNamespaces(final DocumentTypeInfo docTypeInfo,
     rootElement.setAttribute(XMLNS, namespaceUriRoot);
 
     final Map<String, String> map;
-    if (sdkVersion.compareTo(new SdkVersion("1.6")) >= 0) {
+    if (sdkVersion.compareTo(DocumentTypeInfo.TEDEFO_1743_SINCE_SDK_VERSION) >= 0) {
```
Comment (rousso) [inline src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java]: These SINCE_SDK_... stuff is bad practice. We should just remove support of older SDK versions as the application evolves. In the settings we have a list of SDK versions supported by the application. Unsupported versions should be removed from that list and all the SINCE_SDK_ stuff needs to disappear from the code.

**Review by rousso — CHANGES_REQUESTED** [review]

File: src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java (hunk 2 above)
Comment (rouschr) [reply src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java]: Done, I kept only 1.6 as a supported version and remove the version handling code concerning that feature.

Comment (rouschr) [reply src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java]: This breaks some unit tests based on version 1.5 as the code is now trying to sort the XML elements.
I modified the tests and code to use version 1.6 and remove dummy SDK 1.5 test data.
Also the application will now always use the eforms sdk path provided by application.yaml

**Review by rousso — APPROVED** [review]

**Pull request merged**